junit/core/src/main/java/org/jboss/arquillian/junit/State.java

```diff
@@ -88,6 +89,15 @@ static void runnerStarted() {
         lastCreatedRunner.set(lastCreatedRunner.get() + 1);
     }
 
+    static boolean hasAnyArquillianRule(org.junit.runners.model.TestClass testClass) {
```

Comment:
As commented above - maybe we should always check if both are present to fail if not?

Reply:
It's not possible to check it. I will update the documentation with limitation